pkg/plugins/plugins.go

```diff
@@ -109,27 +110,45 @@ func (p *Plugin) activateWithLock() error {
 }
 
 func load(name string) (*Plugin, error) {
-	storage.Lock()
+	return loadWithRetry(name, true)
```

Comment:
Since `load` always means it will retry, does it need to be a separate function?